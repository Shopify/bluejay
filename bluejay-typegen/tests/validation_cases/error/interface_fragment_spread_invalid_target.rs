#[bluejay_typegen::typegen([
    type Query {
        myInterface: MyInterface
    }

    interface MyInterface {
        field: String
    }

    type MyObject implements MyInterface {
        field: String
    }
])]
mod schema {
    #[query([
        {
            myInterface {
                ...MyFragment
            }
        }

        fragment MyFragment on MyObject {
            field
        }
    ])]
    mod query {}
}

fn main() {}
