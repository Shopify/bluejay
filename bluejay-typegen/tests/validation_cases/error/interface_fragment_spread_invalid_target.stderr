error: Error: bluejay_typegen requires fragment spreads on interfaces to target either the interface or one of the interfaces it implements
          ╭─[<unknown>:1:17]
          │
        1 │ { myInterface { ...MyFragment } } fragment MyFragment on MyObject { field }
          │                 ──────┬──────
          │                       ╰──────── Fragment spread on interface type
       ───╯
  --> tests/validation_cases/error/interface_fragment_spread_invalid_target.rs:15:13
   |
15 |       #[query([
   |  _____________^
16 | |         {
17 | |             myInterface {
18 | |                 ...MyFragment
...  |
24 | |         }
25 | |     ])]
   | |_____^
